api/types/accesslist/accesslist.go

```diff
@@ -274,13 +274,27 @@ func (c *CurrentUserAssignments) IsOwner() bool {
 	return c.OwnershipType != accesslistv1.AccessListUserAssignmentType_ACCESS_LIST_USER_ASSIGNMENT_TYPE_UNSPECIFIED
 }
 
+// Option is a functional option for configuring an AccessList.
+type Option func(*AccessList)
+
+// WithSubKind sets the sub-kind of the access list. This is used to differentiate
+func WithSubKind(subKind string) Option {
+	return func(al *AccessList) {
+		al.SubKind = subKind
```

Comment:
You might want to double-check with @espadolini but IIRC the scale team prefers subkind is not used for differentiating between different "types" of objects due to some cache limitations. Can we just have a "type" field in the spec?